rust/arrow/src/array/builder.rs

```diff
@@ -525,7 +515,7 @@ impl<T: ArrowPrimitiveType> ArrayBuilder for PrimitiveBuilder<T> {
                 let sliced = array.buffers()[0].data();
                 // slice into data by factoring (offset and length) * byte width
                 self.values_builder
-                    .write_bytes(&sliced[(offset * mul)..((len + offset) * mul)], len)?;
+                    .write_bytes(&sliced[(offset * mul)..((len + offset) * mul)], len);
```

Comment:
I couldn't reproduce the issue yet, but this line looks a bit suspicious. The first parameter has a larger len (in bytes) than the second `len` parameter indicates (number of `T` elements).

Reply:
Nice 👁️  -- I agree that removing the `len` parameter entirely would be the best course of action here